src/mountscript.sh

```diff
@@ -602,6 +602,28 @@ ensure_aznfswatchdog()
     fi
 }
 
+#
+# To maintain consistency in case of regional account and in general to avoid creating
+# multiple DNAT entries corrosponding to one LOCAL_IP, first check for resolved IP in mountmap.
+# This will help keep mountmap and DNAT entries in sync with each other.
+#
+resolve_ipv4_with_preference_to_mountmap()
+{
+        local fqdn=$1
+        local mountmap_entry=$(grep -m1 "^${fqdn} " $MOUNTMAP)
```

Comment:
we should hold a lock on MOUNTMAP to avoid reading garbled entry if aznfswatchdog is changing it.
This ensures that we either read the mountmap entry before aznfswatchdog changes it (and the dnat rule) or after it, in both cases we can use the entry with the confidence that aznfswatchdog will properly handle the IP address change for this fqdn after this point.